src/main/java/com/elastic/support/diagnostics/commands/DiagVersionCheckCmd.java

```diff
@@ -1,97 +1,100 @@
-            logger.info("Warning: Diagnostic version:{} is not the current recommended release", diagVersion);
-            logger.info("The current release is {}", ver);
-            logger.info("The latest version can be downloaded at {}", downloadUrl);
-
-            Scanner sc = new Scanner(System.in);
-            String feedback = null;
-            boolean valid = false;
-            System.out.println("Continue anyway? [Y/N]");
-            feedback = sc.nextLine();
-            while (!valid) {
-               if ("y".equalsIgnoreCase(feedback) || "n".equalsIgnoreCase(feedback)) {
-                  valid = true;
-               } else {
-                  System.out.println("Continue anyway? [Y/N]");
-                  feedback = sc.nextLine();
-               }
+            // Get the current diagnostic version that was populated in the
+            // manifest generation step - if we're running in
+            // the IDE via a run configuration and/or debugger it will
+            // have a value of "debug" instead of an actual version.
+            String diagVersion = context.getDiagVersion();
+            if (diagVersion.equalsIgnoreCase("debug")) {
+                logger.info("Running in debugger - bypassing check");
+                return;
             }
 
-            if ("n".equalsIgnoreCase(feedback)) {
-               System.out.println("Exiting application");
-               rc = false;
+            RestResult restResult= new RestResult(restClient.exec(ghEndpoint, ghHost, Constants.DEEFAULT_HTTPS_PORT, ghScheme));
+            JsonNode rootNode = JsonYamlUtils.createJsonNodeFromString(restResult.toString());
+            String ver = rootNode.path("tag_name").asText();
+            List<JsonNode> assests = rootNode.findValues("assets");
+            JsonNode asset = assests.get(0);
+            String downloadUrl = asset.path("browser_download_url").asText();
+
+            if (!diagVersion.equals(ver)) {
+                logger.info("Warning: DiagnosticService version:{} is not the current recommended release", diagVersion);
+                logger.info("The current release is {}", ver);
+                logger.info("The latest version can be downloaded at {}", downloadUrl);
+
+                Scanner sc = new Scanner(System.in);
+                String feedback = null;
+                boolean valid = false;
+                System.out.println("Continue anyway? [Y/N]");
+                feedback = sc.nextLine();
+                while (!valid) {
+                    if ("y".equalsIgnoreCase(feedback) || "n".equalsIgnoreCase(feedback)) {
+                        valid = true;
+                    } else {
+                        System.out.println("Continue anyway? [Y/N]");
+                        feedback = sc.nextLine();
+                    }
+                }
+
+                if ("n".equalsIgnoreCase(feedback)) {
+                    System.out.println("Exiting application");
```

Comment:
Actually no, because I want the standard cleanup to occur. There was an exception thrown in the original version, which got lost when my laptop crashed. Will resolve this in a soon to be submitted PR.